app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -159,6 +159,10 @@ def save_hosts_inventory(ems, hashes, target = nil)
           #   and the new hostname is not an ip address
           h[:name] = h[:hostname] if h[:name] =~ ip_part && !(h[:hostname] =~ ip_whole)
 
+          # Update the name to be found.name if it has the same ems_ref and the name
+          # already has a '- int' suffix to work around duplicate hostnames
+          h[:name] = found.name if found.name =~ /#{h[:name]} - \d+$/
```

Comment:
Can you collapse this line and the one above it since they both modify the name?  Since the latter wins, I *think* you could do

```ruby
if found.name =~ /#{h[:name]} - \d+$/
  h[:name] = found.name
elsif h[:name] =~ ip_part && h[:hostname] !~ ip_whole
  h[:name] = h[:hostname]
end
```

Note in my example I also changed `!(a =~ b)` to `a !~ b` because that should be simplified anyway